File: Data Structures & Algorithms/two-integer-sum/submission-6.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        map1 = {}
        for i in range(len(nums)):
            t = target - nums[i]
            if t in map1:
                return [map1[t], i]
            map1[nums[i]] = i

        return []

File: Data Structures & Algorithms/partition-equal-subset-sum/submission-6.java
class Solution {
    public boolean canPartition(int[] nums) {
        int sum = Arrays.stream(nums).sum();
        if(sum % 2 != 0) return false;

        int target = sum / 2;
        return dfs(0, 0, target, nums);
    }

    private boolean dfs(int i, int cur, int target, int[] nums) {
        if(i == nums.length) {
            return cur == target;
        }

        return dfs(i + 1, cur, target, nums) || dfs(i + 1, cur + nums[i], target, nums);
    }
}
